kube_apiserver_metrics/datadog_checks/kube_apiserver_metrics/kube_apiserver_metrics.py

```diff
@@ -0,0 +1,134 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+from copy import deepcopy
+
+from six import iteritems
+from datadog_checks.base import ConfigurationError
+from datadog_checks.checks.openmetrics import OpenMetricsBaseCheck
+
+
+class KubeApiserverMetricsCheck(OpenMetricsBaseCheck):
+    """
+    Collect kubernetes apiserver metrics in the Prometheus format
+    See https://github.com/kubernetes/apiserver
+    """
+    DEFAULT_METRIC_LIMIT = 0
+    # Set up metric_transformers
+    METRIC_TRANSFORMERS = {}
+    DEFAULT_BEARER_TOKEN_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/token'
+    DEFAULT_SCHEME = 'https'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        # Set up metric_transformers
+        self.METRIC_TRANSFORMERS = {
+                'apiserver_audit_event_total': self.apiserver_audit_event_total,
+                'rest_client_requests_total': self.rest_client_requests_total,
+                'apiserver_request_count': self.apiserver_request_count,
+                'apiserver_dropped_requests_total': self.apiserver_dropped_requests_total,
+                'http_requests_total': self.http_requests_total,
+                'authenticated_user_requests': self.authenticated_user_requests,
+        }
+        self.bearer_token_found = False
+        # Create instances we can use in OpenMetricsBaseCheck
+        generic_instances = None
+        if instances is not None:
+            generic_instances = self.create_generic_instances(instances)
+
+        super(KubeApiserverMetricsCheck, self).__init__(
+            name,
+            init_config,
+            agentConfig,
+            instances=generic_instances,
+            default_instances={
+                "kube_apiserver": {
+                    'namespace': 'kube_apiserver',
+                    'metrics': [
+                        {
+                            'apiserver_current_inflight_requests': 'current_inflight_requests',
+                            'apiserver_longrunning_gauge': 'longrunning_gauge',
+                            'go_threads': 'go_threads',
+                            'go_goroutines': 'go_goroutines',
+                            'APIServiceRegistrationController_depth': 'APIServiceRegistrationController_depth',
+                            'etcd_object_counts': 'etcd_object_counts',
+                        }
+                    ],
+                }
+            },
+            default_namespace="kube_apiserver",
+            )
+
+    def check(self, instance):
+        if not self.bearer_token_found:
+            self.log.warn("Not using the service account bearer token file, AuthN/Z will fail.")
+        if not instance.get("prometheus_url"):
+            raise ConfigurationError("Missing prometheus_endpoint field. Skipping check")
```

Comment:
This already happens in https://github.com/DataDog/integrations-core/blob/073b120631ee91a578161d9d8ce04ebb2d1b534b/datadog_checks_base/datadog_checks/base/checks/openmetrics/base_check.py#L50-L53